types/src/front/lib/core_api.ts

```diff
@@ -670,35 +670,42 @@ export class CoreAPI {
     return this._resultFromResponse(response);
   }
 
-  async getDataSourceDocuments({
-    dataSourceId,
-    limit,
-    offset,
-    projectId,
-    viewFilter,
-  }: {
-    dataSourceId: string;
-    limit: number;
-    offset: number;
-    projectId: string;
-    viewFilter?: CoreAPISearchFilter | null;
-  }): Promise<
+  async getDataSourceDocuments(
+    {
+      dataSourceId,
+      documentIds,
+      projectId,
+      viewFilter,
+    }: {
+      dataSourceId: string;
+      documentIds?: string[];
+      projectId: string;
+      viewFilter?: CoreAPISearchFilter | null;
+    },
+    pagination?: { limit: number; offset: number }
```

Comment:
`nit`: could we use `OffsetPaginationParams` here?

Reply:
It lives in front for now, but will move it in the follow up PR.